sisl/geom/nanoribbon.py

```diff
@@ -154,3 +159,461 @@ def zgnr(width, bond=1.42, atoms=None):
+                with an open configuration). If you pass an imaginary number on an invalid
+                situation, an error will be raised.
+
+                If not provided, it defaults to 1.
+            shift: int, optional
+                The shift of this section with respect to the previous one.
+
+                The exact meaning of the value depends on the sections that are
+                being joined so that only valid ribbons are produced:
+                    - If both ribbons are odd, ribbons are aligned on the center
+                    and then shifted:
+                        ^ (shift) zig zag atoms if the incoming section is thinner than the
+                            previous one.
+                        ^ (2 * shift) zig zag atoms if they differ in a multiple of 4 atoms
+                            and both configurations are closed or if they differ in (2 + a
+                            multiple of 4 atoms) and configurations are different (one open
+                            and one closed.)
+                        ^ (1 + 2 * shift) zig zag atoms otherwise.
+                    - If both ribbons are even, they are aligned by the open end of the last
+                    section, not the incoming one, and then they are shifted:
+                        ^ (0) zig zag atoms if the shift is 0. In this case both sections are 
+                        forced to have the same edge open.
+                        ^ (-1 + 2 * abs(shift)) zig zag atoms in the appropiate direction if abs(shift) is
+                        bigger than 1. Notice that you can only shift the open edges toward the center.
+                    - If the ribbons have different parities, they are aligned on the open edge
+                    of the even ribbon and then shifted:
+                        ^ (0) zig zag atoms if shift is 0 in the particular cases that (i) the incoming
+                        odd section is thinner than the last even section or (ii) the last section, 
+                        being odd, has a border in the open configuration. Only in those cases does it make
+                        sense to exactly match the edges.
+                        ^ (bias + 2 * abs(shift)) zig zag atoms in the appropiate direction, where `bias` is
+                        either -1 or 1 depending on the exact situation and always ensures that the minimum
+                        shift is always `abs(shift) == 1`.
+
+                If not provided, it defaults to 0.
+            atoms: Atom, optional
+                Value to pass to the `atoms` argument of `nanoribbon`. If not provided,
+                it defaults to the `atoms` argument passed to this function.
+    bond: float
+        The bond length of the ribbon.
+    atoms: Atom, optional
+        The default atom to use for the sections. For each section, if it contains an `atoms`
+        value, it will be used instead of this one.
+    kind: {'armchair', 'zigzag'}, optional
```

Comment:
when you have a set, you don't need `optional`, it is implicit

Reply:
Really? Why is that? An argument can be positional but only accept values within a set, doesn't it?

Reply:
I meant for the documentation:

```
kind: {'armchair', 'zigzag'}
```
nothing more, this is described in the numpy-docs manual.

Reply:
Ok, I've changed it. But I don't understand the logic. I will read the numpy-docs manual :)